src/writeData/subscriptions/components/JsonPathInput.tsx

```diff
@@ -143,8 +154,14 @@ const JsonPathInput: FC<Props> = ({name, formContent, updateForm, itemNum}) => {
                   active={active}
                   onClick={onClick}
                   testID={`${tagType}-json-parsing-type`}
+                  status={
+                    edit ? ComponentStatus.Default : ComponentStatus.Disabled
+                  }
                 >
-                  {dataType}
+                  {tagType
+                    ? sanitizeType(formContent.jsonTagKeys[itemNum].type)
+                    : sanitizeType(formContent.jsonFieldKeys[itemNum].type) ??
+                      dataType}
```

Comment:
is the `dataType` default just supposed to apply to the `sanitizeType(formContent.jsonFieldKeys[itemNum].type)` statement and not the tags one?

Reply:
no, good catch